wgsl/index.bs

```diff
@@ -395,21 +411,103 @@ The following types are <dfn dfn noexport>host-shareable</dfn>:
 * [[#array-types]] if the array type has a stride attribute and its element type is host-shareable
 * [[#struct-types]] if all its members are host-shareable
 
+The contents of uniform buffers and storage buffers must be of host-shareable type.
+
 TODO: *This is a stub*: Collectively, the *stride* and *offset* attributes are called *layout attributes*.
 
 ### Storage Classes TODO ### {#storage-class}
 
+Memory locations are partitioned into <dfn noexport>storage classes</dfn>.
+Each storage class has unique properties affecting mutability, visibility, and the values they may contain.
+
+
+<table class='data'>
+  <thead>
+    <tr><td>Storage class
+        <td>Readable by shader?<br>Writable by shader?
+        <td>Sharing among invocations
+        <td>Scope
+        <td>Restrictions on stored values
+        <td>Notes
+  </thead>
+  <tr><td>input
+      <td>Read-only
+      <td>Same invocation only
+      <td>[=Module scope=]
+      <td>[=IO-shareable=]
+      <td>Contains values written by an upstream pipeline stage, or by the implementation.
+  <tr><td>output
+      <td>Read-write
+      <td>Same invocation only
+      <td>[=Module scope=]
+      <td>[=IO-shareable=]
+      <td>Shader writes values for downstream pipeline processing.
```

Comment:
Maybe

```
Output for a downstream pipeline stage. Zero initialized if required.
```

Reply:
I'll take the first part. 

The zero-initialized aspect is already described in "Variable and const" https://gpuweb.github.io/gpuweb/wgsl.html#variable-declaration